lib/netsuite_integration/inventory_adjustment.rb

```diff
@@ -1,210 +1,224 @@
+        @adjustment = NetSuite::Records::InventoryAdjustment.new
+        adjustment.external_id = adjustment_id
+        adjustment.memo = adjustment_memo
+        adjustment.tran_date = NetSuite::Utilities.normalize_datetime_to_netsuite(adjustment_date.to_datetime)
 
-            adjustment.account={internal_id: adjustment_account_id}
-            if adjustment_department.present?
-                adjustment.department={internal_id: adjustment_dept_id}
-            end
-            adjustment.adj_location={internal_id: adjustment_location}
-            location={internal_id: adjustment_location}
-            adjustment.inventory_list=build_item_list
-            #we can sometimes receive adjustments were everything is zero!
-            if adjustment.inventory_list.inventory.present?
-                adjustment.add
-                if adjustment.errors.any?{|e| "WARN" != e.type}
-                    raise "Adjustment create failed: #{adjustment.errors.map(&:message)}"
-                else
-                    line_item = { adjustment_id: adjustment_id, netsuite_id: adjustment.internal_id, description: adjustment_memo,type: 'Adjustment' }
-                    if transfer_order?
-                        ExternalReference.record :transfer_order, adjustment_id, { netsuite: line_item }, netsuite_id: adjustment.internal_id
-                    elsif register_sale?
-                        ExternalReference.record :register_sale, adjustment_identifier, { netsuite: line_item }, netsuite_id: adjustment.internal_id
-                    elsif sales_inv_adjustment?
-                        ExternalReference.record :sales_inv_adjustment, adjustment_identifier, { netsuite: line_item }, netsuite_id: adjustment.internal_id
-                    else
-                        ExternalReference.record :inventory_adjustment, adjustment_id, { netsuite: line_item }, netsuite_id: adjustment.internal_id
-                    end
-                end
+        adjustment.account = { internal_id: adjustment_account_id }
+        if adjustment_department.present?
+          adjustment.department = { internal_id: adjustment_dept_id }
+        end
+        adjustment.adj_location = { internal_id: adjustment_location }
+        adjustment.inventory_list = build_item_list
+        # we can sometimes receive adjustments were everything is zero!
+        if adjustment.inventory_list.inventory.present?
+          adjustment.add
+          if adjustment.errors.any? { |e| e.type != 'WARN' }
+            raise "Adjustment create failed: #{adjustment.errors.map(&:message)}"
+          else
+            line_item = { adjustment_id: adjustment_id,
+                          netsuite_id: adjustment.internal_id,
+                          description: adjustment_memo,
+                          type: 'Adjustment' }
+            if transfer_order?
+              ExternalReference.record :transfer_order, adjustment_id,
```

Comment:
@arosenk35 why is it sometimes `adjustment_id` and sometimes `adjustment_identifier` (like on line 189)?

Reply:
adjustment_id is used is used for ns external reference and has a different for to the id registered within cangaroo external xref tables.  

adjustment id has to stay the same format at all costs it is used to verify that the txn has not been processed within NS ...... The biggest issue with adjustments is that we have adjustments coming in from various entry points  .eg. sales/transfer orders/receipts/returns ...... and the identifier for these txn's are very different.  ad we want to guarantee the NS id format... identifier is not that important

Reply:
ok, thanks for clarifying. Just a confusing name!